package.json

```diff
@@ -44,7 +46,12 @@
     "nodemon": "^2.0.4",
     "nyc": "^15.1.0",
     "prettier": "^2.0.5",
+    "sinon": "^9.0.2",
+    "sinon-chai": "^3.5.0",
     "ts-node": "^8.10.2",
     "typescript": "^3.9.7"
+  },
+  "dependencies": {
+    "abortable-iterator": "^3.0.0"
```

Comment:
Are we using `abortable-iterator`? I can't find it in the code, is it a peer of something?

Reply:
Thanks for notice it. I tried to use it first to solve the problem, until I realized I could solve it in a better way just taking advantage of the native iterator contract